common/src/main/java/com/wynntils/wc/utils/lootrun/LootrunUtils.java

```diff
@@ -284,6 +266,23 @@ private static void renderPoints(
         }
     }
 
+    private static BlockValidness checkBlockValidness(Level level, ColoredPoint point) {
+        BlockValidness state = BlockValidness.INVALID;
+        Iterable<BlockPos> blocks = getBlocksForPoint(point);
+
+        for (BlockPos blockInArea : blocks) {
+            BlockState blockStateInArea = level.getBlockState(blockInArea);
+            if (blockStateInArea.is(Blocks.BARRIER)) {
+                state = BlockValidness.HAS_BARRIER;
+            } else if (blockStateInArea.getCollisionShape(level, blockInArea) != null) {
+                state = BlockValidness.VALID;
+                return state;
+            }
+        }
+
+        return state;
```

Comment:
this should just be inlined as `BlockValidness.INVALID`

Reply:
If we don't auto return HAS_BARRIER it shouldn't.

Reply:
ah